core/src/main/java/org/ehcache/core/config/DefaultConfiguration.java

```diff
@@ -77,7 +78,7 @@ public DefaultConfiguration(ClassLoader classLoader, ServiceCreationConfiguratio
   public DefaultConfiguration(Map<String, CacheConfiguration<?, ?>> caches, ClassLoader classLoader, ServiceCreationConfiguration<?>... services) {
     this.services = unmodifiableCollection(Arrays.asList(services));
     this.caches = new ConcurrentHashMap<String, CacheConfiguration<?, ?>>(caches);
-    this.classLoader = classLoader;
+    this.classLoader = classLoader == null ? ClassLoading.getDefaultClassLoader() : classLoader;
```

Comment:
That's what I am saying. But I also suggest refactoring the constructors so that you end up having the check in a single one. And similarly the fact that cache config map is in a concurrent map and that service configs is an immutable collection.
Makes sense?